app/controllers/Application.scala

```diff
@@ -1,17 +1,15 @@
 package controllers
 
+import play.api.Configuration
 import play.api.mvc._
-import play.api.Play
-
-class Application extends Controller {
+import play.api.routing.Router
 
+class Application(configuration: Configuration, router: => Router) extends Controller {
```

Comment:
I would perhaps add a comment about the call-by-name `=>` as it's non-obvious but took us a while to figure out